src/main/java/groots/sad/project/MassTransitSystem/EventSimulator.java

```diff
@@ -0,0 +1,94 @@
+package groots.sad.project.MassTransitSystem;
+
+import groots.sad.project.MassTransitSystem.entity.*;
+import groots.sad.project.MassTransitSystem.manager.BusManager;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+@Component
+public class EventSimulator {
+
+
+    private static BusManager busManager = BusManager.getInstance();
+    private static List<BusStop> busStops = new ArrayList<>();
+    private static List<Route> routes = new ArrayList<>();
+    private static List<Event> eventQueue = new ArrayList<>();
+    private static EventSimulator eventSimulator = new EventSimulator();
+
+    private EventSimulator() {
+
+        busManager = BusManager.getInstance();
+        busStops = new ArrayList<>();
+        routes = new ArrayList<>();
+        eventQueue = new ArrayList<>();
+    }
+
+    public static EventSimulator getInstance(){
+        return eventSimulator;
+    }
+
+    private static void readData(String fileName) {
+
+        final String DELIMITER = ",";
+        try {
+            Scanner takeCommand = new Scanner(new File(fileName));
+            String[] tokens;
+            do {
+                String userCommandLine = takeCommand.nextLine();
+                tokens = userCommandLine.split(DELIMITER);
+                switch (tokens[0]) {
+                    case "add_stop":
+                        Location location = new Location(Double.valueOf(tokens[4]), Double.valueOf(tokens[5]));
```

Comment:
we should do error handling for accessing tokens using index if in case an erroneous input is given. for eg: if length of tokens is 4 and we are checking tokens[5]. This error scenario should be handled.

Reply:
i had planned to do all the validations later. i have planned to first make the happy path running and then handle the corner or destructive scenarios. what do you think?